modules/images/image-loading-optimization/storage/post-type.php

```diff
@@ -152,12 +146,12 @@ static function ( $url_metric ) use ( $trigger_error ) {
  * @since n.e.x.t
  * @access private
  *
- * @param string $url                  URL for the URL metrics. This is used purely as metadata.
- * @param string $slug                 URL metrics slug (computed from query vars).
- * @param array  $validated_url_metric Validated URL metric. See JSON Schema defined in ilo_register_endpoint().
+ * @param string         $url            URL for the URL metrics. This is used purely as metadata.
+ * @param string         $slug           URL metrics slug (computed from query vars).
+ * @param ILO_URL_Metric $new_url_metric New URL metric.
  * @return int|WP_Error Post ID or WP_Error otherwise.
  */
-function ilo_store_url_metric( string $url, string $slug, array $validated_url_metric ) {
+function ilo_store_url_metric( string $url, string $slug, ILO_URL_Metric $new_url_metric ) {
```

Comment:
See my above comment: I think a function like this could be made more intuitive if `ILO_URL_Metric` covered the URL and slug as well. This way you'd only pass around the object, not the other two values in addition.